openwisp_monitoring/monitoring/static/monitoring/js/chart-utils.js

```diff
@@ -36,6 +43,73 @@ django.jQuery(function ($) {
       }, initDateRangePickerWidget);
       initDateRangePickerWidget(start, end);
 
+    function handleChartZoomChange(chartsContainers) {
+      // Handle chart zooming with custom dates
+      var zoomCharts = document.getElementsByClassName(chartsContainers);
+      for (var zoomChart of zoomCharts) {
+        if (!zoomChart) {
+          localStorage.setItem(isChartZoomed, false);
+          localStorage.setItem(isCustomDateRange, false);
+          return;
+        }
+        zoomChart.on('plotly_relayout',
+          function (eventdata) { // jshint ignore:line
+            var eventEnd = eventdata['xaxis.range[1]'];
+            var eventStart = eventdata['xaxis.range[0]'];
+            if (!eventEnd || !eventStart) {
```

Comment:
what case is this? Do you care to clarify with a comment? Is this when the chart gets dragged?

Reply:
When the chart is zoomed out, this code will be executed, and I'll add a comment for it.